Sources/MeiliSearch/Model/SearchParameters.swift

```diff
@@ -82,15 +82,57 @@ public struct SearchParameters: Codable, Equatable {
         SearchParameters(query: value)
     }
 
-   private func commaRepresentation(_ array: [String]) -> String {
-        array.joined(separator: ",")
+    enum CodingKeys: String, CodingKey {
+        case query = "q"
+        case offset
+        case limit
+        case attributesToRetrieve
+        case attributesToCrop
+        case cropLength
+        case attributesToHighlight
+        case filters
+        case facetFilters
+        case facetsDistribution
+        case matches
     }
 
-    private func commaRepresentationEscaped(_ array: [String]) -> String {
-        var value: String = "["
-        value += array.map({ string in "\"\(string)\"" }).joined(separator: ",")
-        value += "]"
-        return value
+}
+
+extension SearchParameters {
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(query, forKey: .query)
+        if limit != 20 {
```

Comment:
Why should we need to do a condition with the default value?

Reply:
This is a filter to default parameters, so far I know these values are fixed in the server and if they are not informed in the HTTP request the server will apply the default ones, am I correct?

Reply:
The API is based in this documentation: https://docs.meilisearch.com/references/search.html#search-in-an-index-with-post-route

Note how the first example only contains the query item as mandatory, all other options are optional. This script remove the need of informing the default values.

Reply:
But what is the point of doing this?

> they are not informed in the HTTP request the server will apply the default ones,

Yes, exactly, so no need to check the default value. Just send to MeiliSearch what the users fill, we don't care if they fill default values. MeiliSearch will manage that.

This big `if` conditions are really complex for a job that MeiliSearch already does.

Or maybe I missed something!